test/core/utils/respondable.js

```diff
@@ -1,515 +1,561 @@
 
+    it('throws when targeting a window that is not a frame in the page', function() {
+      var blankPage = window.open('');
+      var frameCopy = window.open(frameWin.location.href);
+      // Cleanup
       setTimeout(function() {
-        done();
-      }, 100);
+        blankPage.close();
+        frameCopy.close();
+      });
+
+      assert.throws(function() {
+        respondable(blankPage, 'greeting');
+      });
+      assert.throws(function() {
+        respondable(frameCopy, 'greeting');
+      });
     });
 
-    it('should reject messages that do not have a uuid', function(done) {
-      eventData = {
-        _respondable: true,
-        _source: 'axeAPI.' + axe.version,
-        message: 'Help us Obi-Wan',
-        _axeuuid: 'otherAxe'
-      };
+    it('is not triggered by "repeaters"', function(done) {
+      var calls = 0;
+      frameSubscribe('greeting', function() {
+        calls++;
+      });
+      // Repeat fire the event
+      frameWin.addEventListener('message', function handler(evt) {
+        frameWin.postMessage(evt.data, '*');
+        frameWin.removeEventListener('message', handler);
+      });
 
-      win = {
-        postMessage: function() {
-          event.data = JSON.stringify(eventData);
-          document.dispatchEvent(event);
-        }
-      };
+      respondable(frameWin, 'greeting', 'hello');
+      setTimeout(
+        captureError(function() {
+          assert.equal(calls, 1);
+          done();
+        }, done),
+        100
+      );
+    });
+  });
 
-      axe.utils.respondable(win, 'Death star', null, true, function() {
-        done(new Error('should not call callback'));
+  describe('respond', function() {
+    it('passes the response back', function(done) {
+      var receivedResponse;
+      frameSubscribe(
+        'greeting',
+        captureError(function(message, keepalive, respond) {
+          assert.isFalse(keepalive);
+          respond({ greet: 'bonjour' });
+        }, done)
+      );
+
+      respondable(frameWin, 'greeting', 'hello', false, function(message) {
+        receivedResponse = message;
       });
 
-      setTimeout(function() {
-        done();
-      }, 100);
+      afterMessage(
+        window,
+        captureError(function() {
+          assert.deepEqual(receivedResponse, { greet: 'bonjour' });
+          done();
+        }, done)
+      );
     });
 
-    it('should reject messages that do not have a matching uuid', function(done) {
-      eventData = {
-        _respondable: true,
-        _source: 'axeAPI.' + axe.version,
-        message: 'Help us Obi-Wan',
-        _axeuuid: 'otherAxe'
-      };
-
-      win = {
-        postMessage: function(message) {
-          var data = JSON.parse(message);
-          eventData.uuid = data.uuid + 'joker tricks!';
-          event.data = JSON.stringify(eventData);
-          document.dispatchEvent(event);
-        }
-      };
+    it('prohibits multiple response calls with keepalive: false', function(done) {
```

Comment:
This test was confusing since the assert shows keepalive is false. Maybe update to reflect where it's set?

```suggestion
    it('prohibits multiple response calls when respond sets keepalive to false', function(done) {
```